packages/composites/token-allocation/src/config.ts

```diff
@@ -1,20 +1,43 @@
 import { util } from '@chainlink/ea-bootstrap'
 import { getDefaultConfig } from '@chainlink/external-adapter'
 import { getDataProvider } from './dataProvider'
-import { Config } from './types'
+import { Config, SourceRequestOptions } from './types'
+import { adapters } from './dataProvider'
 
 export const DEFAULT_TOKEN_DECIMALS = 18
 export const DEFAULT_TOKEN_BALANCE = 1
 
+export const ENV_DATA_PROVIDER_URL = 'DATA_PROVIDER_URL'
+
+export const getURL = (prefix: string, required = false) =>
+  required
+    ? util.getRequiredEnv(ENV_DATA_PROVIDER_URL, prefix)
+    : util.getEnv(ENV_DATA_PROVIDER_URL, prefix)
+
 export const makeConfig = (prefix = ''): Config => {
-  const dataProviderUrl = util.getRequiredEnv('DATA_PROVIDER_URL', prefix)
-  const defaultConfig = getDefaultConfig(prefix)
-  defaultConfig.api.baseURL = defaultConfig.api.baseURL || dataProviderUrl
-  defaultConfig.api.method = 'post'
+  const sources: SourceRequestOptions = {}
+
+  for (const a of adapters) {
+    const name = a.NAME.toLowerCase()
+    const url = getURL(name)
+    if (url) {
+      const defaultConfig = getDefaultConfig(prefix)
+      defaultConfig.api.baseURL = url
+      defaultConfig.api.method = 'post'
+      sources[name] = getDataProvider(defaultConfig.api)
+    }
+  }
 
   return {
-    priceAdapter: getDataProvider(defaultConfig.api),
+    sources,
     defaultMethod: util.getEnv('DEFAULT_METHOD', prefix) || 'price',
     defaultQuote: util.getEnv('DEFAULT_QUOTE') || 'USD',
   }
 }
+
+export const makeOptions = ({ sources }: Config) => {
```

Comment:
Do we really need this method, and can we do this in `makeConfig`?
From the user perspective/api, it seems confusing to both have `makeOptions` & `makeConfig`.

Reply:
Yeah I see what you mean. I'll return it as part of the config.